packages/contracts-bedrock/src/deploy-utils.ts

```diff
@@ -0,0 +1,298 @@
+import assert from 'assert'
+
+import { ethers, Contract } from 'ethers'
+import { Provider } from '@ethersproject/abstract-provider'
+import { Signer } from '@ethersproject/abstract-signer'
+import { sleep, awaitCondition, getChainId } from '@eth-optimism/core-utils'
+import { HttpNetworkConfig } from 'hardhat/types'
+
+export const deployAndVerifyAndThen = async ({
+  hre,
+  name,
+  args,
+  contract,
+  iface,
+  postDeployAction,
+}: {
+  hre: any
+  name: string
+  args: any[]
+  contract?: string
+  iface?: string
+  postDeployAction?: (contract: Contract) => Promise<void>
+}) => {
+  const { deploy } = hre.deployments
+  const { deployer } = await hre.getNamedAccounts()
+
+  const result = await deploy(name, {
+    contract,
+    from: deployer,
+    args,
+    log: true,
+    waitConfirmations: hre.deployConfig.numDeployConfirmations,
+  })
+
+  await hre.ethers.provider.waitForTransaction(result.transactionHash)
+
+  if (result.newlyDeployed) {
+    if (!(await isHardhatNode(hre)) && hre.network.config.live !== false) {
+      // Verification sometimes fails, even when the contract is correctly deployed and eventually
+      // verified. Possibly due to a race condition. We don't want to halt the whole deployment
+      // process just because that happens.
+      try {
+        console.log('Verifying on Etherscan...')
+        await hre.run('verify:verify', {
```

Comment:
We should double check that this works with the foundry setup by doing a testnet deployment, was getting some weird errors at one point that didn't make sense so i made a hh task for verifications using our foundry setup and it works